vm_supervisor/snapshots.py

```diff
@@ -29,9 +34,13 @@ def __init__(self, path: Path, algorithm: SnapshotCompressionAlgorithm):
     def delete(self) -> None:
         self.path.unlink(missing_ok=True)
 
-    async def upload(self) -> ItemHash:
-        # TODO: Upload snapshots to Aleph Network
-        pass
+    async def upload(self, vm_hash: ItemHash) -> ItemHash:
+        account = get_fallback_account()
+        async with AuthenticatedAlephClient(account=account, api_server="https://official.aleph.cloud") as client:
+            message, status = await client.create_store(
+                file_path=self.path, storage_engine=StorageEnum.ipfs, sync=True, ref=f"snapshot_{vm_hash}"
+            )
+            assert status == MessageStatus.PROCESSED
```

Comment:
I think it's important here to delete old snapshots from the network as well. A call to `client.forget()`would make a lot of sense when we delete the previous snapshot. We'll implement some depth to this (keep the last N snapshots) later.

Reply:
Good point, thanks, I just Implemented it.